server/src/routes/services.js

```diff
@@ -23,80 +24,106 @@ router.get('/', (req, res) => {
-                    location: req.body.location,
-                });
-                newService
-                    .save()
-                    .then(response => {
-                        return res.status(200).json({
-                            success: true,
-                            result: response,
+                }).then(service => {
+                    if (service)
+                        return res.status(422).json({
+                            errors: [{ service: 'service already exists' }],
                         });
-                    })
-                    .catch(err => {
-                        return res.status(500).json({
-                            errors: [{ error: err }],
+                    else {
+                        const newService = new Service({
+                            provider: req.body.username,
+                            name: req.body.name,
+                            description: req.body.description,
+                            cost: req.body.cost,
+                            duration: req.body.duration,
+                            availability: req.body.availability,
+                            location: req.body.location,
                         });
-                    });
+                        newService
+                            .save()
+                            .then(response => {
+                                return res.status(200).json({
+                                    success: true,
+                                    result: response,
+                                });
+                            })
+                            .catch(err => {
+                                return res.status(500).json({
+                                    errors: [{ error: err }],
+                                });
+                            });
+                    }
+                });
+            } else {
+                return res.status(500).json({
+                    errors: [{ error: 'user not found' }],
+                });
             }
-        }
-    );
+        })
+        .catch(err => {
+            return res.status(500).json({
+                errors: [{ error: err }],
+            });
+        });
 });
 
 /* UPDATE a service. Returns the OLD object */
 router.put('/:service_id', (req, res) => {
     //can't change the username attached to service
     if (Object.prototype.hasOwnProperty.call(req.body, 'username')) {
-        return res.status(500).json({ error: 'Cannot change user in service' });
+        return res
+            .status(500)
+            .json({ errors: 'Cannot change user in service' });
     }
 
     //update the service based on req.body
     Service.findByIdAndUpdate(
         req.params.service_id,
         req.body,
         (err, service) => {
-            if (err) return res.status(500).json({ error: err });
+            if (err) return res.status(500).json({ errors: err });
```

Comment:
why change this to errors?

Reply:
errors is the correct key name, to be consistent with our responses everything should have errors : []